File: first_unrepeated_char.rb
# http://programmingpraxis.com/2013/04/30/first-unrepeated-character-in-a-string/

def find_first_unrepeated_char(string)
  (0...string.length).each {|i| return string[i] if string.count(string[i]) == 1}
end

p find_first_unrepeated_char("aabbcddd")

File: first_unrepeated_char.pl
#!/usr/bin/env perl
# http://programmingpraxis.com/2013/04/30/first-unrepeated-character-in-a-string/
use strict;
use warnings;

sub first_unrepeated_char {
  my $string = shift;
  for (my $i = 0; $i < length $string; $i++) {
    my $char = substr($string, $i, 1);
    my $count = 0;
    for (my $j = $i+1; $j < length $string; $j++) {
      $count++ if $char eq substr($string, $j, 1);
    }
    for (my $k = $i-1; $k >= 0; $k--) {
      $count++ if $char eq substr($string, $k, 1);
    }
    return $char if $count == 0;
  }
  return "not found"
}

my $result = first_unrepeated_char("aabbcddd");
print "$result\n";
